src/gui/qgsdatasourceselectdialog.h

```diff
@@ -68,6 +69,13 @@ class GUI_EXPORT QgsDataSourceSelectDialog: public QDialog, private Ui::QgsDataS
      */
     void setLayerTypeFilter( QgsMapLayerType layerType );
 
+    /**
+     * Sets a description label
```

Comment:
Can you also add a note about the position in the dialog where this description will be shown?
```suggestion
     * Sets a description label for the original data source
     * \note the description will be shown at the bottom of the dialog
```

Reply:
@elpaso , sure can add a \note here, I'm glad to receive feedbacks from this dialog's creator :smile: 

As for the 'for the original data source' addition, I won't append that since the description label is meant to be slightly more generic than only specifying the original data source. For e.g., if one day we use this data source selection dialog within the context of processing algorithms, that description placeholder might be used to describe a parameter from which a data source is being sought, etc.